store/src/db/sql.rs

```diff
@@ -125,16 +155,83 @@ pub fn select_accounts_by_block_range(
-    accounts: &[(AccountId, RpoDigest)],
+    accounts: &[(AccountId, Option<AccountDetails>, RpoDigest)],
     block_num: BlockNumber,
 ) -> Result<usize> {
-    let mut stmt = transaction.prepare("INSERT OR REPLACE INTO accounts (account_id, account_hash, block_num) VALUES (?1, ?2, ?3);")?;
+    let mut acc_stmt = transaction.prepare(
+        "INSERT OR REPLACE INTO accounts (account_id, account_hash, block_num) VALUES (?1, ?2, ?3);",
+    )?;
+    let mut select_details_stmt = transaction.prepare(
+        "SELECT nonce, vault, storage, code FROM account_details WHERE account_id = ?1;",
+    )?;
+    let mut new_details_stmt = transaction.prepare(
+        "INSERT INTO account_details (account_id, nonce, vault, storage, code) VALUES (?1, ?2, ?3, ?4, ?5);"
+    )?;
+    let mut update_details_stmt = transaction.prepare(
+        "UPDATE account_details SET nonce = ?2, vault = ?3, storage = ?4 WHERE account_id = ?1;",
+    )?;
 
     let mut count = 0;
-    for (account_id, account_hash) in accounts.iter() {
-        count +=
-            stmt.execute(params![u64_to_value(*account_id), account_hash.to_bytes(), block_num])?
+    for (account_id, details, account_hash) in accounts.iter() {
+        let account_id = *account_id;
+        count += acc_stmt.execute(params![
+            u64_to_value(account_id),
+            account_hash.to_bytes(),
+            block_num
+        ])?;
+        if let Some(details) = details {
+            match details {
+                AccountDetails::Full(account) => {
+                    debug_assert!(account.is_new());
+                    debug_assert_eq!(account_id, u64::from(account.id()));
+                    let inserted = new_details_stmt.execute(params![
+                        u64_to_value(account.id().into()),
+                        u64_to_value(account.nonce().as_int()),
+                        account.vault().to_bytes(),
+                        account.storage().to_bytes(),
+                        account.code().to_bytes(),
+                    ])?;
+
+                    debug_assert_eq!(inserted, 1);
+                },
+                AccountDetails::Delta(delta) => {
+                    let mut rows = select_details_stmt.query(params![u64_to_value(account_id)])?;
+                    let Some(row) = rows.next()? else {
+                        return Err(DatabaseError::AccountNotOnChain(account_id));
+                    };
+
+                    let mut account = Account::new(
+                        account_id.try_into()?,
+                        AssetVault::read_from_bytes(row.get_ref(1)?.as_blob()?)?,
+                        AccountStorage::read_from_bytes(row.get_ref(2)?.as_blob()?)?,
+                        AccountCode::read_from_bytes(row.get_ref(3)?.as_blob()?)?,
+                        column_value_as_u64(row, 0)?
+                            .try_into()
+                            .map_err(DatabaseError::CorruptedData)?,
+                    );
```

Comment:
Request: use `select_account_details` instead

Reply:
This code was removed later.